misc/synchronization.md

```diff
@@ -0,0 +1,413 @@
+an exception. Note that because of C++ limitations we cannot propagate the correct exception type and
+message to tiles. Therefore they will contain a generic exception.
+
+## Matrix Views
+
+A matrix view can be can be used to create "sub-DAG" in the global DAG of a matrix.
+It gives the chance to take control of a particular asynchronous step of the matrix,
+allowing to concurrently schedule the work that will be executed after the work
+scheduled by the matrix view has been executed and the matrix view is notified that
+all the tasks has been scheduled (`done()`, `doneWrite()`, destructor).
+This is very handy, because it allows to schedule tasks inside the sub-DAG of the matrix view,
+and it will be like inserting new dependencies in between two already existing ones.
+
+A matrix view (`dlaf::matrix::MatrixView`) can be created from a `Matrix` or from another `MatrixView`,
+which will be indicated as "parent matrix".
+Both constant and non-constant views are available. The main difference is that `operator()`
+is not allowed for constant matrices as only read-only tasks can be performed on its tiles.
+
+When a matrix view is created each successive call of `operator()` and `read()` on the parent matrix
+will return futures which depend on the tasks scheduled with the futures obtained by the matrix view.
+It is possible to transfer the scheduling of all or just a subset of the tiles.
+Currently it can be specified with the `uplo` parameter if:
+- all the tiles or
+- only the tiles which contain an element of the diagonal or the upper (lower) triangular
+
+has to be transferred to the new view.
+
+In this way the DAG scheduling can be performed concurrently on the parent matrix and the matrix view,
+but the correct dependencies are guarenteed. An important fact to be noted is that the view needs to be
+able to notify that no other tasks will be scheduled to enable the next tasks created from the parent matrix.
+The `MatrixView` API contains 3 ways to accomplish that:
+- the destructor notifies that no new tasks will be scheduled on any of the tile,
+- `done(tile_index)` notifies that no new task will be scheduled on the tile with the given index
+(After this call any use of `operator()` or `read()` with the same tile index will return an error.),
+- `doneWrite(tile_index)` notifies that no new read-write task will be scheduled on the tile with the given index
+(After this call any use of `operator()` with the same tile index will return an error.).
+
+It has to be noted that read-only tasks of the parent matrix and of the view can be performed at the same time
+if no read-write tasks has been scheduled in between (see [examples](#view-examples)).
+
+### View Examples
+
+(Note: this examples are used to illustrate how dependencies are determined,
+however they cannot be compiled since they are simplified to improve readability.
+(e.g. constructors and template parameters are omitted)).
```

Comment:
```suggestion
however they cannot be compiled since they are simplified to improve readability
(e.g. constructors and template parameters are omitted).)
```